ddtelemetry/tests/flock.rs

```diff
@@ -0,0 +1,70 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+#![cfg(unix)]
+
+use std::{
+    io::{self, Read, Write},
+    os::unix::net::UnixStream,
+    time::Duration,
+};
+
+use ddtelemetry::ipc::platform::locks::FLock;
+use spawn_worker::{assert_child_exit, entrypoint, fork::set_default_child_panic_handler, Stdio};
+use tempfile::tempdir;
+
+static ENV_LOCK_PATH: &str = "__LOCK_PATH";
+
+#[no_mangle]
+pub extern "C" fn flock_test_entrypoint() {
+    set_default_child_panic_handler();
+    let lock_path = std::env::var(ENV_LOCK_PATH).unwrap();
+
+    let _l = FLock::try_rw_lock(lock_path).unwrap();
+    let mut c: UnixStream = spawn_worker::recv_passed_fd().unwrap().into();
+
+    c.write_all(&[0]).unwrap(); // signal readiness
+    let mut buf = [0; 10];
+    assert!(c.read(&mut buf).unwrap() > 0); // wait for signal to closepp
+
+    std::process::exit(0); // exit without explicitly freeing
+}
+
+#[test]
+fn test_file_locking_works_as_expected() {
```

Comment:
can the tests (and test entrypoints) be put in a `#[cfg(test)]` block to not get compiled on release?

Reply:
doesn't need to be. This file is compiled separately as an integration test. It will not get into release